### DISCUSSION TITLE
Compress files using Vue3, Nuxt3, Webpack5

### DISCUSSION THREAD

**Discussion by ChrisChristidis** [nuxt/nuxt#19597] [Questions]
I have the code below in my nuxt.config.ts and after build and run as production environment, the compression doesn't work properly for all my files.

In nuxt.config.ts

// "compression-webpack-plugin": "^10.0.0",
const CompressionPlugin = require("compression-webpack-plugin");

export default defineNuxtConfig({
  ...,
  webpack: {
      plugins: [
        ...,
        new CompressionPlugin()
      ],
  },
});

I import css in every single page, layout and component like:

async beforeMount() {
      await import(
        /* webpackChunkName: "my-name-css" */
        /* webpackExports: ["default", "named"] */ "@/assets/css/Common/TopArea/TopBarImage/onLoadDesktop.css"
      );
}

and the gzip compression doesn't work for all of these files.

![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/117259563/224478933-adcb68d2-279c-4c28-8986-f4ee1c9bb3c5.png)

Do you have any idea how to fix it?

Thanks in advance.

**Comment by vagelis-chantzis** [comment]
Same issue here, is there anyone who came across and addressed it?

**Comment by TheAlexLichter** [comment]
You can use [Nitro's `compressPublicAssets` setting](https://nitro.unjs.io/config#compresspublicassets)

**Reply by ChrisChristidis** [reply]
I added this to nuxt.config.ts
nitro: {
    compressPublicAssets: {
      gzip: true,
    },
  },

But the same issue and also it compressed them as "br" some of these files.
![MicrosoftTeams-image](https://user-images.githubusercontent.com/117259563/225290604-4135e5b9-1a69-4bb6-8c17-c13c9892b09e.png)

**Reply by TheAlexLichter** [reply]
Be aware that compression might not be worth applying for small files already (time to decompress > transmission time saved). Brotli (br) is better than GZIP or DEFLATE so that's correct.

**Reply by ChrisChristidis** [reply]
ok, thank you for the explanation. Some files above of that list are gzip and some are br. But for the files that are not compressed what can I do?

**Reply by TheAlexLichter** [reply]
Probably they are not worth compressing and stay uncompressed (eg. not worth applying Brotli nor GZIP for 364b)

**Reply by ChrisChristidis** [reply]
Ok, so you recommend "nitro" instead of "compression-webpack-plugin", although there are less files being compressed, because it seems to be more worthy?

**Reply by TheAlexLichter** [reply]
Nitro should handle compression as needed for static assets, yes. But also, your reverse proxy or platform should handle compression for you besides that ☺️ See https://github.com/unjs/nitro/issues/1007
